dicthash/dicthash.py

```diff
@@ -110,19 +114,19 @@ def _generate_string_from_dict(d, blacklist, whitelist, prefix=''):
     ----------
     d : dict
         Dictionary to be converted
-    blacklist : list
+    blacklist : List[Hashable], optional
         List of keys to exclude from conversion. Blacklist overrules
         whitelist, i.e., keys appearing in the blacklist will
         definitely not be used.
-    whitelist: list
+    whitelist: list, optional
         List of keys to include in conversion.
     """
     if whitelist is None:
         whitelist = list(d.keys())
     if blacklist is not None:
-        whitelist = set(whitelist).difference(blacklist)
+        whitelist = list(set(whitelist).difference(blacklist))
     # Sort whitelist according to the keys converted to str
-    if len(whitelist) > 0:
+    if len(whitelist) > 0 and whitelist is not None:
```

Comment:
isn't the `not None` implicit in `if len() > 0`? maybe we could write this as `if whitelist`?

Reply:
Nevermind, this served to fix an error thrown by `mypy` but eventually I fixed this with the `@overload` construction (see below). I reverted this change.